### ISSUE TITLE
Calculation with arbitrary units

### ISSUE THREAD

**Issue by tylkomat** [angularsen/UnitsNet#744]
I'm developing a learning application for the topic of technical 
mechanics and came across your UnitsNet package. In our application, 
which is targeting scholars and early students, players have to find 
values and put them into a specific calculator. For example a 
Force Calculator. We want to allow putting any value into the 
calculator, not just mass and acceleration. Is there a way to achieve 
that with UnitsNet? Currently it throws an Error.

Here is a small example: https://dotnetfiddle.net/vMqZZL

We don't have many formulars, maybe UnitsNet is overkill for this task: 
![formulas](https://user-images.githubusercontent.com/815915/73524571-ae6f1300-440d-11ea-9e7d-ddc553fc143e.png)

**Comment by angularsen** [comment]
Hi, I don't fully understand your question and how you want to use Units.NET.

From your dotnetfiddle it seems you are asking if you can multiply `Length` and `Acceleration`, or `Length` and `Mass`? Yes you would get an error for those. All arithmetic must be explicitly implemented between quantities where it makes sense and where the resulting quantity exists in Units.NET. 

Or are you asking whether it is possible to multiply any two kinds of quantity? Something like `IQuantity * IQuantity = IQuantity`? If so, the answer is no. Dynamic arithmetic is currently not supported. Operator overloads must be explicitly implemented for each combination of two quantity types, which gives strongly typed results.

Or are you asking if Units.NET can parse and solve equations like `50 km / 1 h` that should result in a `Speed` quantity with value `50 km/h`? If so, no it only supports parsing individual quantities like `Length.Parse("50 km")`, but it cannot parse or solve equations.

Hope this helps.

**Comment by tylkomat** [comment]
Yes basically the first two. Parsing is not necessary the formulas would be hardcoded. It's just that the parameters can be of different types. From what I understood now I would have to implement the Fake Datatypes and all combinations of possible calculations.

Thanks for the information and fast response.

**Comment by angularsen** [comment]
Yes, in the current shape of Units.NET, I believe you will have to implement something yourself.

We have discussed dynamic computations with `IQuantity` before, but it requires some work to support something like that and someone with a real need for it should be the one to implement it.

Some previous discussions:
https://github.com/angularsen/UnitsNet/issues/592
https://github.com/angularsen/UnitsNet/issues/514
https://github.com/angularsen/UnitsNet/issues/681

I'd be interested to learn how you solve it though and if maybe your solution is something that can be added to the library.

I'll close this for now. Reopen or create a new issue to continue the discussion.

Best, Andreas

**Comment by tylkomat** [comment]
I found this one and and it looks promising: https://www.codeproject.com/Articles/1066008/Using-physical-quantities-and-units-of-measurement

**Comment by angularsen** [comment]
Nice find, I just skimmed the article but it does seem to cover the ideas we are looking for here. The source code brings some .dlls so it seems not all the source code is available, but I was able to browse things like `Unit` and how they add and multiply quantities of different units/dimensions.

It would be awesome to bring something similar into Units.NET, but it needs to be championed by someone who actually needs/wants it and who can put in the effort required designing, implementing and writing tests as well as consuming the functionality in a real life scenario to make sure it actually works.
